src/agent.h

```diff
@@ -148,6 +148,11 @@ struct juice_agent {
 
 	thread_t resolver_thread;
 	bool resolver_thread_started;
+
+	/* This is a custom field that can be used to store user data for the agent.
+	 * Use the juice_agent_set_data and juice_agent_get_data methods to store and retrieve the data.
+	 */
+	void* user_ptr;
```

Comment:
What is the difference with the existing `user_ptr` in `juice_config_t`?

Reply:
Hey, I wanted each agent to have its own (unique) data.


My current implementation uses the same `juice_config_t` for all the agents.
I just realised that if I create multiple instances of `juice_config_t`, each having its own unique data, then each agent will also have its own user pointer.
I'll close the PR, thanks

Reply:
Oh OK, the config is actually copied for each agent, so you can change it between call to `juice_create()`, it won't interfere with already-created agents.